### ISSUE TITLE
Some questions about training.

### ISSUE THREAD

**Issue by crashkk** [neuroailab/TDANN#9]
Hi, it's a really interesting work! I wonder that whether the topography of each layer can emerge while training from scratch instead of training from a initialized pretrained network through task+spatial loss? And what about the performance or topography emergence in other much smaller datasets(cifar10 etc.)?

**Comment by eshedmargalit** [comment]
Thanks for your interest.

> I wonder that whether the topography of each layer can emerge while training from scratch instead of training from a initialized pretrained network through task+spatial loss

The weights of the models in the paper are trained from scratch, there is no pretraining of model weights before the final training step. The _spatial positions_ of the model units are pre-optimized as described, and we did find that this pre-optimization was necessary to give good performance. There must be some choice of the (x, y) coordinates of each model unit before training, and randomly assigning units to spatial positions doesn't work as well. This wasn't surprising to us, since the positions of real neurons in the brain are not random when an animal is born -- the brain does a lot of "pre-optimization" of neuron positions before visual experience begins.

> And what about the performance or topography emergence in other much smaller datasets(cifar10 etc.)?
We didn't try CIFAR10, but supplemental figure 6 (pasted here) shows results for two natural image datasets (ImageNet and Ecoset) and two simpler datasets (white noise and sine gratings). We found that training on naturalistic datasets was necessary to get the right topography. This is probably because it's hard to develop selectivity for categories like faces if the training set doesn't have faces. 

<img width="878" alt="Image" src="https://github.com/user-attachments/assets/786fff22-4b61-49b4-8c44-e55dfb8b88de" />

It would be fun for somebody to try training one of our networks on a smaller dataset and report back what happens!

**Comment by crashkk** [comment]
> The weights of the models in the paper are trained from scratch, there is no pretraining of model weights before the final training step. The _spatial positions_ of the model units are pre-optimized as described, and we did find that this pre-optimization was necessary to give good performance. There must be some choice of the (x, y) coordinates of each model unit before training, and randomly assigning units to spatial positions doesn't work as well. This wasn't surprising to us, since the positions of real neurons in the brain are not random when an animal is born -- the brain does a lot of "pre-optimization" of neuron positions before visual experience begins.

Thank you for such a timely response. I also found that when the coordinate of each unit in features of size (C,H,W) was just randomly placed in a 2d cortical sheet, the model actually failed to learn the spatial information and the spatial loss couldn't go down successfully.

**Comment by crashkk** [comment]
> We didn't try CIFAR10, but supplemental figure 6 (pasted here) shows results for two natural image datasets (ImageNet and Ecoset) and two simpler datasets (white noise and sine gratings). We found that training on naturalistic datasets was necessary to get the right topography. This is probably because it's hard to develop selectivity for categories like faces if the training set doesn't have faces.

Thanks! I get it now

**Comment by eshedmargalit** [comment]
Thanks for sharing, that result makes sense! I'll close this issue for now but let me know if you have any other questions or ideas that you'd like to share.
